webapp/src/ts/modules/reports/reports.component.ts

```diff
@@ -533,6 +534,10 @@ export class ReportsComponent implements OnInit, AfterViewInit, OnDestroy {
     }
 
     const isMaxReportsSelected = this.selectedReports?.length >= this.LIMIT_SELECT_ALL_REPORTS;
-    return isMaxReportsSelected || this.reportsList?.length === this.selectedReports?.length;
+    const check = isMaxReportsSelected || this.reportsList?.length === this.selectedReports?.length;
+    if(check){
+      this.isSelectingAllReports = false
+    }
+    return check;
```

Comment:
Please fix linting here: 
```suggestion
    const finishedSelecting = isMaxReportsSelected || this.reportsList?.length === this.selectedReports?.length;
    this.isSelectingAllReports = !finishedSelecting;
    return finishedSelecting;
```

This also has a renaming suggestion and reduces code depth.